src/OrderFormAcceptanceTests.TestData/SupplierInfo.cs

```diff
@@ -59,48 +29,43 @@ FROM CatalogueItem
 
-            return SqlExecutor.Execute<string>(connectionString, query, new { supplierId }).FirstOrDefault();
-        }
+            var result = await SqlExecutor.ExecuteAsync<SupplierDetails>(connectionString, query, new { supplierId });
 
-        public static int SupplierWithMoreThanOneSolution(string connectionString)
-        {
-            var query =
-                $@"SELECT SupplierId, COUNT(*) FROM CatalogueItem WHERE CatalogueItemTypeId = {(int)CatalogueItemType.Solution} GROUP BY SupplierId ORDER BY 2 DESC";
-
-            return SqlExecutor.Execute<int>(connectionString, query, null).FirstOrDefault();
+            return result.Single();
         }
 
-        public static string GetSupplierSolutionNameWithPrice(string connectionString, string supplierId)
+        public static async Task<bool> GetSupplierWithContactDetails(string connectionString, string supplierName)
         {
-            var query =
-                $@"SELECT ci.[Name] FROM dbo.CatalogueItem AS ci INNER JOIN CataloguePrice AS cp ON cp.CatalogueItemId = ci.CatalogueItemId WHERE ci.CatalogueItemTypeId = 1 AND ci.PublishedStatusId = 3 AND ci.SupplierId = @supplierId;";
-
-            return SqlExecutor.Execute<string>(connectionString, query, new { supplierId }).FirstOrDefault();
-        }
+            var query = @"SELECT Id
+                            FROM SupplierContact
+                            WHERE SupplierId = (SELECT Id FROM Supplier AS s WHERE s.[Name] = @supplierName);";
 
-        public static IEnumerable<string> GetPublishedCatalogueItems(string connectionString, string supplierId, CatalogueItemType itemType)
-        {
-            var query = $@"SELECT Name FROM dbo.CatalogueItem WHERE SupplierId = @supplierId AND PublishedStatusId = 3 AND CatalogueItemTypeId = @itemType;";
+            var result = (await SqlExecutor.ExecuteAsync<Guid?>(connectionString, query, new { supplierName })).SingleOrDefault();
 
-            return SqlExecutor.Execute<string>(connectionString, query, new { supplierId, itemType });
+            if (result is null)
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
         }
 
-        private static IEnumerable<SupplierDetails> SupplierLookup(string connectionString, CatalogueItemType catalogueItemType, ProvisioningType provisioningType)
+        private static IEnumerable<SupplierDetails> SupplierLookup(string connectionString, CatalogueItemType catalogueItemType, Domain.ProvisioningType provisioningType)
         {
-            var query = @"SELECT ci.[SupplierId], su.[Name]      
+            var query = @"SELECT ci.[SupplierId], su.[Name], su.Address    
                             FROM [dbo].[CatalogueItem] ci
                             INNER JOIN CataloguePrice pr ON ci.CatalogueItemId=pr.CatalogueItemId
                             INNER JOIN Supplier su On ci.SupplierId=su.Id
```

Comment:
Recommend including `AS` for table aliases, for example:

```sql
FROM dbo.CatalogueItem AS ci
```

Would recommend removing square brackets (except for `Name`) and adding spaces around operators. However, these are just non-essential suggestions that I think would improve readability – can always be done later as tech debt.